hs_core/tests/api/views/test_aggregation_zip_download.py

```diff
@@ -118,3 +120,25 @@ def _test_aggregation_download_zip(self, folder, raster_file, auto_aggregate=Tru
         # verify the download path
         download_path = response_json["download_path"]
         self.assertTrue(download_path.endswith(f"{aggr_main_file_path}.zip"))
+
+        aggr_main_file_path = aggr_main_file.get_short_path()
+        url_params = {'shortkey': self.resource.short_id, 'filepath': aggr_main_file_path}
+        url = reverse('get_resource_file', kwargs=url_params)
+        url = f"{url}/?zipped=True&aggregation=True"
+
+        request = self.factory.get(url)
+        request.user = self.user
+        self.add_session_to_request(request)
+        response = file_download_url_mapper(request, shortkey=self.resource.short_id, filepath=aggr_main_file_path)
+        # check that the download request was successful
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_json = json.loads(response.content.decode())
+        # check that a task id was generated
+        task_id = response_json["task_id"]
+        self.assertTrue(len(task_id) > 0, msg='ensure a task_id is returned for async zipping')
+        # verify the zip creation succeeded
+        sleep(10)
```

Comment:
consider a loop to check the status of the task, instead of a "raw" sleep?

Reply:
I mean, I guess we are only calling `_test_aggregation_download_zip` a handful of times so sleeping 10 on each one is at most going to add ~1min to our overall test execution time. No need for a loop...